workflows/references/main.nf

```diff
@@ -72,81 +79,76 @@ workflow REFERENCES {
-            .map { meta, fasta, fai, fa_sizes ->
-                return fai[0][0] && fa_sizes[0][0] ? null : [meta, fasta]
-            }
+    BWAMEM2_INDEX(
+        input.bwamem2_fasta.map { meta, file ->
+            return file ? [meta, file] : null
+        }
+    )
 
-        generate_sizes = tools.split(',').contains('sizes')
-        SAMTOOLS_FAIDX(fasta_samtools, [[id: 'no_fai'], []], generate_sizes)
+    DRAGMAP_HASHTABLE(
+        input.dragmap_fasta.map { meta, file ->
+            return file ? [meta, file] : null
+        }
+    )
 
+    GATK4_CREATESEQUENCEDICTIONARY(
+        input.createsequencedictionary_fasta.map { meta, file ->
+            return file ? [meta, file] : null
+        }
+    )
+
+    SAMTOOLS_FAIDX(
+        input.fasta_samtools.map { meta, file ->
+            return file ? [meta, file] : null
+        },
+        [[id: 'no_fai'], []],
+        tools.contains('sizes')
+    )
+
+    // TODO: be smarter about input assets
+    //   Here we either mix+GT an empty channel (either no output or no input faidx) with the faidx return faidx
+    //   And we filter out the empty value
+    faidx = input.fasta_fai
+        .mix(SAMTOOLS_FAIDX.out.fai)
+        .groupTuple()
+        .map { meta, file ->
+            return file[1] ? [meta, file[1]] : [meta, file]
+        }
 
-        // TODO: be smarter about input assets
-        //   Here we either mix+GT an empty channel (either no output or no input faidx) with the faidx return faidx
-        //   And we filter out the empty value
-        faidx = input.fasta_fai
-            .mix(SAMTOOLS_FAIDX.out.fai)
-            .groupTuple()
-            .map { meta, fai ->
-                return fai[1] ? [meta, fai[1]] : [meta, fai]
-            }
+    // TODO: be smarter about input assets
+    //   Here we either mix+GT an empty channel (either no output or no input sizes) with the sizes return sizes
+    //   And we filter out the empty value
+    sizes = input.fasta_sizes
+        .mix(SAMTOOLS_FAIDX.out.sizes)
+        .groupTuple()
+        .map { meta, file ->
+            return file[1] ? [meta, file[1]] : [meta, file]
+        }
 
-        // TODO: be smarter about input assets
-        //   Here we either mix+GT an empty channel (either no output or no input sizes) with the sizes return sizes
-        //   And we filter out the empty value
-        sizes = input.fasta_sizes
-            .mix(SAMTOOLS_FAIDX.out.sizes)
-            .groupTuple()
-            .map { meta, file ->
-                return file[1] ? [meta, file[1]] : [meta, file]
-            }
+    faidx_intervals = faidx
+        .mix(input.intervals)
+        .groupTuple()
+        .map { meta, file ->
+            return file[1] ? null : [meta, file]
+        }
 
-        versions = versions.mix(SAMTOOLS_FAIDX.out.versions)
-    }
+    BUILD_INTERVALS(faidx_intervals, [])
 
-    if (tools && tools.split(',').contains('gffread')) {
+    if (tools.contains('gffread')) {
         GFFREAD(input.gff, [])
```

Comment:
Why didn't you continue with the simplification of the if statements ?

Reply:
Just testing it out on some processes, I didn't want to modify everything in one go as I just wanted to tackle intervals on this PR.

Reply:
That does make sense.
Could you open an issue in this case in the repository to have a follow up for standardization ?